File: backend/Commands.py
from enum import Enum

# commands are in the form of
# <cmd>:<target>:<payload>
# regexp for recognition: '.+|.+|.+'
class Commands(str, Enum):
    LOGIN = 'login'
    LOGOUT = 'logout'
    MESSAGE = 'message'

    LEFTROLL = "left"
    RIGHTROLL = "right"
    THRUST = "thrust"

class Targets(str, Enum):
    CONTROLLER = 'controller'
    CONTROLLER1 = "controller1"
    CONTROLLER2 = "controller2"
    CONTROLLER3 = "controller3"
    CONTROLLER4 = "controller4"
    MASTER = "master"
    SERVER = "server"
    PLAYER = "player"

class Defaults(str, Enum):
    NONE = 'none'
    DELIMETER = '|'
    TARGET_DELIMETER = ','

class Payloads(str, Enum):
    JOINED = "joined"
    SIGNUP = 'signup'

File: frontend/src/commands.js
/**
 * Created by daniel on 4/6/17.
 */
export const commands = {
    LOGIN : "login",
    LOGOUT : "logout",
    SIGNUP : "signup",
    MESSAGE : "message",

    LEFTROLL : "left",
    RIGHTROLL : "right",
    THRUST : "thrust",
    NONE : "none",

    REQUEST : 'request',
    ANSWER : 'answer',

    READY : 'ready'
};


export const targets = {
    CONTROLLER : "controller",
    MASTER : "master",
    SERVER : "server",
};

export const playerStates = {
    ALIVE : 1,
    DEAD : 0,
    FINISHED: 2,
};

export const colors = [
    0xff0000,
    0x00ff00,
    0xff00ff,
    0x0000ff
];



export const payloads = {
    JOINED : "joined",
    SIGNUP : 'signup',
    FAILED : 'failed',

    LISTROOMS : 'listRooms'
}

export const DELIMETER = "|";
export const TARGET_DELIMETER = ":";
export const NONE = "none";
export const NEW = "new";
